### ISSUE TITLE
Add some sort of Installation instructions

### ISSUE THREAD

**Issue by rocky** [bioperl/Bio-Root#2]
In other words, instructions to  configure, install and use the module.

**Comment by fjossandon** [comment]
Rocky, since Chris mentioned that issue notifications from Git are not going to the mailing list, I would like to mention that BioPerl have a mailing list where devs are suscribed:
http://lists.open-bio.org/mailman/listinfo/bioperl-l
You can suscribe or send emails directly to bioperl-l@mailman.open-bio.org, for wider audience.

**Comment by rocky** [comment]
This is a little off topic, but...

If Chris starts watching the BioPerl projects he his interested in &mdash; click the watch button or go to URL https://github.com/bioperl/Bio-Root/subscription &mdash; he will get emails. 

In my view, this has got to be better than automatically "subscribing" everybody on that list to anything that goes on here. That said, if that's the route you want to go (and again I think it inadvisable), then see https://developer.github.com/webhooks/ .

**Comment by hlapp** [comment]
@rocky: Chris (assuming you mean @cjfields) is already watching this and other repos. However, BioPerl is much more than one or a few lead developers. There's a community of several hundreds. This community is interacting on the mailing list, not the Github issue tracker, and although Github issue tracker threads have worked amazingly well for some communities for some purposes, they don't for every community for every purpose that needs communication or deliberation.

**Comment by rocky** [comment]
Ok. But in my opinion this issue &mdash; adding install instructions to the Bio::Root git repository &mdash; is _not_ something I think several hundred people are interested in. 

If I got email about every issue for every github project under bioperl, it would shy me away from joining the list.

**Comment by rocky** [comment]
Maybe there's a misunderstanding. This issue is about adding installation instructions for this repository only.  Every project should have some sort of installation instructions.

**Comment by hlapp** [comment]
> If I got email about every issue for every github project under bioperl, it would shy me away from joining the list.

Of course. But you did lead into this with the statement that "this is a little off-topic". So I was assuming you did not mean your suggestion to apply to this particular or similar small issues, but to all issues, small or large.

**Comment by majensen** [comment]
I think spending a little time on the mailing list before articulating a large list of desired things on GitHub would be advisable for an enthusiastic person wanting to develop some goodwill with a large, ~18 year old community, in order to make a positive and lasting contribution.

**Comment by rocky** [comment]
Guys -  There has been more time devoted to making explanations and off-topic suggestions by, _several_ people than it would have taken to just fix the issue. 

How long does it take to write install instructions? If you don't know the _proper_ way to do that, ok. But if you do, it shouldn't take more than 10 minutes, at least for a first rough draft. There has clearly been more time spent skirting around the issue than solving it. 

If you want to develop good will, how about just fixing some of the _simple_ issues.

**Comment by hlapp** [comment]
@rocky I don't know what the install instructions are, otherwise I would indeed have written them. I suspect the same goes for @majensen. I don't know on which basis you infer that I spend time skirting around the issue.

**Comment by rocky** [comment]
Now I am now really confused. You are listed as the (or a) _owner_ of this project and you don't know what steps someone should take to install  Bio::Root?

**Comment by cjfields** [comment]
There is some historical context missing here. Bio-Root hasn't been released officially yet as it is intended to be split out as a separate dist.  Uploading it to CPAN is not an option yet, and having an official way to install it outside of CPAN (eg README) is premature until we discuss it on list to have a path forward for the next developer version (1.7).

We can initiate that but I can't participate until I'm back from vacation, on the 30th. Kinda bad timing to start this right now ;)

Sent from a Phone of Ingenuity (+3)

-------- Original message --------
From: "R. Bernstein"
Date:12/23/2014 12:49 PM (GMT-06:00)
To: bioperl/Bio-Root
Cc: "Fields, Christopher J"
Subject: Re: [Bio-Root] Add some sort of Install instructions (#2)

Now I am now really confused. You are listed as the (or a) owner of this project and you don't know what steps someone should use to install Bio::Root?

**Comment by cjfields** [comment]
As an aside, the effort to push this forward had stagnated a bit (I'm the sole dev making releases at the moment and I have a full time job), so any discussion that drives things forward constructively is a good thing.  And branches, PR, etc like this are more than welcome.

-------- Original message --------
From: "R. Bernstein"
Date:12/23/2014 12:49 PM (GMT-06:00)
To: bioperl/Bio-Root
Cc: "Fields, Christopher J"
Subject: Re: [Bio-Root] Add some sort of Install instructions (#2)

Now I am now really confused. You are listed as the (or a) owner of this project and you don't know what steps someone should use to install Bio::Root?

**Comment by cjfields** [comment]
Ah, had assumed this was a PR.  We can add something here that would be more helpful but I think we need to make sure to mention (until we have a CPAN version) that git installs aren't fully supported yet, unless you have the entire dzil toolchain.

-------- Original message --------
From: "R. Bernstein"
Date:12/23/2014 12:49 PM (GMT-06:00)
To: bioperl/Bio-Root
Cc: "Fields, Christopher J"
Subject: Re: [Bio-Root] Add some sort of Install instructions (#2)

Now I am now really confused. You are listed as the (or a) owner of this project and you don't know what steps someone should use to install Bio::Root?

**Comment by majensen** [comment]
Rocky, as a [former IBM Researcher, Chaitin co-author and 30 year developer](http://www.meetup.com/LispNYC/events/47373722/), you are probably aware that sometimes things in free software repositories are broken, and sometimes it takes a long time for both interest and free time to co-occur in one or more people to fix things. It turns out that the deconstruction of BioPerl into reasonably sized and logical chunks is (at least to me) harder than it looks on the outside. I actually have a Build.PL for Bio::Root in my local repo. It's part of a fairly complex effort to make [bioperl-run](https://github.com/bioperl/bioperl-run/tree/topic/issue11) actually installable. But those changes actually need some input from the community before we just commit. The people who are interested will read the posts, and those who ain't won't. No big deal. 
And if you would like to fix the issue, by all means code away and make a PR. If you want to just use BioPerl, then clone the [release tag](https://github.com/bioperl/bioperl-live/tree/release-1-6-924).

**Comment by rocky** [comment]
Thanks, everyone, for the explanations. A pity it took this much verbiage to get to the bottom of what's going on and why this isn't easily moved forward.

Yes, free software repositories are sometimes broken. Given my experience so far with this project, I think it best to wait for you all  &mdash; owners, collaborators, and discussion groups &mdash; to sort it out.

No rush. Happy Holidays!
